contract_payment_mode/models/contract.py

```diff
@@ -17,6 +17,8 @@ class ContractContract(models.Model):
     @api.depends("partner_id", "contract_type")
     def _compute_payment_mode_id(self):
         for rec in self:
+            if rec.payment_mode_id:
+                continue
```

Comment:
IMO this is incorrect, reviewing what was commented in https://github.com/OCA/contract/pull/1212#pullrequestreview-2871182601 I think that if the problem is in `contract_sale_payment_mode` that is where something like this should be done (in  `_compute_payment_mode_id()` method from `contract_sale_payment_mode`):

```
_self = self - self.filtered(lambda x: not x.payment_mode_id)
super(ContractContract, _self)._compute_payment_mode_id()
```

What do you think @pedrobaeza ?

Reply:
For me, one thing is clear: if you change the `contract_type` from `purchase` to `sale`, whatever payment mode was selected, it's incorrect, as you are now doing the inverse operation (inbound<>outbund).

Reply:
The payment mode should be reset thus.